src/Compilers/VisualBasic/Portable/Binding/InitializerSemanticModel.vb

```diff
@@ -97,12 +97,12 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                         Debug.Assert(initializer IsNot Nothing)
                         If initializer.Kind = SyntaxKind.EqualsValue Then
                             Dim enumSymbol = DirectCast(Me.MemberSymbol, SourceEnumConstantSymbol)
-                            boundInitializer = binder.BindFieldAndEnumConstantInitializer(enumSymbol, DirectCast(initializer, EqualsValueSyntax), isEnum:=True, diagnostics:=diagnostics, constValue:=Nothing)
+                            boundInitializer = WrapWithEqualsValue(binder.BindFieldAndEnumConstantInitializer(enumSymbol, DirectCast(initializer, EqualsValueSyntax), isEnum:=True, diagnostics:=diagnostics, constValue:=Nothing), initializer)
```

Comment:
My thinking above is incorrect--the cases that produce things other than expressions also need to be covered. I'll add some tests for things like "As New" declarations, and expect to have to rework this code again.